sympy/stats/tests/test_rv.py

```diff
@@ -385,3 +386,9 @@ def test_issue_6810():
     assert P(Eq(Y, 0)) == 0
     assert P(Or(X > 2, X < 3)) == 1
     assert P(And(X > 3, X > 2)) == S(1)/2
+
+def test_issue_20286():
+    n, p = symbols('n p')
+    B = Binomial('B', n, p)
+    assert H(B) == Sum(Piecewise((-p**_k*(1 - p)**(-_k + n)*log(p**_k*(1 - p)**(-_k + n)*binomial(n, _k))*binomial(n, _k), (_k >= 0) & (_k <= n)), (nan, True)), (_k, 0, n))
```

Comment:
Directly copying and comparing `Sum` will not work. What you need to do is to create, `k = Dummy(k, integer=True)` and use it in RHS.

Reply:
Ok, will do so, thanks.

Reply:
I tried doing this, I am getting an AssertionError.

Reply:
```python
In [1]: from sympy.stats import *

In [2]: n, p = symbols('n p')

In [3]: B = Binomial('B', n, p)

In [4]: k = Dummy('k', integer=True)

In [5]: H(B)
Out[5]: 
  n                                                                      
_____                                                                    
╲                                                                        
 ╲                                                                       
  ╲   ⎧  k        -k + n    ⎛ k        -k + n ⎛n⎞⎞ ⎛n⎞                   
   ╲  ⎪-p ⋅(1 - p)      ⋅log⎜p ⋅(1 - p)      ⋅⎜ ⎟⎟⋅⎜ ⎟  for k ≥ 0 ∧ k ≤ n
   ╱  ⎨                     ⎝                 ⎝k⎠⎠ ⎝k⎠                   
  ╱   ⎪                                                                  
 ╱    ⎩                      nan                            otherwise    
╱                                                                        
‾‾‾‾‾                                                                    
k = 0                                                                    

In [6]: print(H(B))
Sum(Piecewise((-p**_k*(1 - p)**(-_k + n)*log(p**_k*(1 - p)**(-_k + n)*binomial(n, _k))*binomial(n, _k), (_k >= 0) & (_k <= n)), (nan, True)), (_k, 0, n))

In [7]: entr = Sum(Piecewise((-p**k*(1 - p)**(-k + n)*log(p**k*(1 - p)**(-k + n)*binomial(n, k))*binomial(n, k), (k >= 0) & (k <= n)), (nan, True)), (k, 0, n))

In [8]: entr.dummy_eq(H(B))
Out[8]: True
```